typing/typedecl.ml

```diff
@@ -1714,6 +1810,25 @@ let report_error ppf = function
         "cannot be checked"
   | Val_in_structure ->
       fprintf ppf "Value declarations are only allowed in signatures"
+  | Invalid_native_repr_attribute_payload kind ->
+      fprintf ppf "Payload of [@%s ] attribute must be \
+                   either a type either nothing"
+        (string_of_native_repr_kind kind)
+  | Multiple_native_repr_attributes ->
+      fprintf ppf "Too many [@unboxed ]/[@untagged ] attributes"
+  | Cannot_unbox_or_untag_type Unboxed ->
+      fprintf ppf "Don't know how to unbox this type. Only float, int32, \
+                   int64 and nativeint can be unboxed"
+  | Cannot_unbox_or_untag_type Untagged ->
+      fprintf ppf "Don't know how to untag this type. Only int \
+                   can be untagged"
+  | Type_does_not_match_native_repr_attribute_payload kind ->
+      fprintf ppf "Type and payload of [@%s: ] have different %s \
```

Comment:
In case a type is abstract, the payload is supposed to be the real type. So it can be any type. But maybe allowing the user to put a type here is a bad idea after all. I'll remove this part for now